Core/Src/main.c

```diff
@@ -87,17 +89,46 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  uint8_t RxData[3];
+  uint8_t TxData[3] = {0x01, 0x80, 0x00};
+  uint16_t ADC_Value;
+  float PWM_DutyCycle; //
+  uint8_t PWM_DutyCycle_Count;
+  const uint16_t PERIOD = 0xfa00; // 64000, our period, in hexadecimal
```

Comment:
Honestly, I think it makes more sense to use decimal here. Sometimes I ask people to use hex to represent the data because those data are highly associated with bytes, using hexadecimal can increase the readability of the text. Here, I don't think using hex is offering any readability advantage comparing to others...